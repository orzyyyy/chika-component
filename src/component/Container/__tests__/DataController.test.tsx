import React from 'react';
import { shallow, mount } from 'enzyme';
import 'nino-cli/scripts/setup';
let DataController;
switch (process.env.LIB_DIR) {
  case 'lib':
    DataController = require('../../../../lib/component/Container/DataController')
      .default;
    break;
  case 'dist':
    DataController = require('../../../../dist/lib/Container').default;
    break;
  default:
    DataController = require('../DataController').default;
    break;
}

describe('DataController', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <DataController tableId={-2} menuId={-2}>
        <>
          <div data-key="pjnm">1</div>
          <div data-key="dam_cd">2</div>
        </>
      </DataController>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('handlePowerStr should work correctly', () => {
    const wrapper = mount(
      <DataController tableId={-2} menuId={-2}>
        <>
          <div data-key="pjnm">1</div>
          <div data-key="dam_cd">2</div>
        </>
      </DataController>,
    ).instance();
    expect(wrapper.handlePowerStr('Select')).toEqual({
      select: true,
      delete: false,
      update: false,
      add: false,
    });
    expect(wrapper.handlePowerStr('Select,Add')).toEqual({
      select: true,
      delete: false,
      update: false,
      add: true,
    });
    expect(wrapper.handlePowerStr('Select,Add,Del')).toEqual({
      select: true,
      delete: true,
      update: false,
      add: true,
    });
    expect(wrapper.handlePowerStr('Select,Add,Del,Update')).toEqual({
      select: true,
      delete: true,
      update: true,
      add: true,
    });
  });

  it('getPrimaryKey should work correctly', () => {
    const wrapper = mount(
      <DataController tableId={-2} menuId={-2}>
        <>
          <div data-key="pjnm">1</div>
          <div data-key="dam_cd">2</div>
        </>
      </DataController>,
    ).instance();
    expect(wrapper.getPrimaryKey([{ fname: 'test', iskey: true }])).toBe(
      'test',
    );
  });
});
