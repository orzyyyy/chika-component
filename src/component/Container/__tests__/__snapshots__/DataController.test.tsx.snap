// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataController render correctly 1`] = `
<div
  className="DataController"
>
  <ContainerCore
    config={Array []}
    dataSource={Array []}
    domain="."
    formatControls={[Function]}
    loading={true}
    menuId={-2}
    onDelete={[Function]}
    onMapPickerChange={[Function]}
    onSearch={[Function]}
    onSort={[Function]}
    power={
      Object {
        "add": false,
        "delete": false,
        "select": false,
        "update": false,
      }
    }
    primaryKey=""
    tableId={-2}
    total={0}
    updatePageChange={[Function]}
    updatePageForm={Array []}
    updatePageMapBoxItem={
      Object {
        "key": "key",
        "lat": "-1",
        "lng": "-1",
      }
    }
    updatePageSave={[Function]}
  >
    <div
      data-key="pjnm"
    >
      1
    </div>
    <div
      data-key="dam_cd"
    >
      2
    </div>
  </ContainerCore>
</div>
`;
